### PR TITLE
adds aarch64-darwin to jameica

### PR DESCRIPTION
**Pull request by fbettag**
adds aarch64-darwin to jameica

### CODE REVIEW

**Review by 4evy — CHANGES_REQUESTED** [review]
Adding `aarch64-darwin` to the `meta.platforms` isn't enough; you need to extend `swtSystem` by adding a check for `aarch64-darwin` so it can copy the native jar file instead of using the `x86_darwin`, which requires Rosetta

This is what you will need to do:

```diff
diff --git a/pkgs/applications/office/jameica/default.nix b/pkgs/applications/office/jameica/default.nix
index 32cbcf088c51..adb3d91c828b 100644
--- a/pkgs/applications/office/jameica/default.nix
+++ b/pkgs/applications/office/jameica/default.nix
@@ -24,6 +24,7 @@ let
     else if stdenv.hostPlatform.system == "x86_64-linux" then "linux64"
     else if stdenv.hostPlatform.system == "aarch64-linux" then "linux-arm64"
     else if stdenv.hostPlatform.system == "x86_64-darwin" then "macos64"
+    else if stdenv.hostPlatform.system == "aarch64-darwin" then "macos-aarch64"
     else throw "Unsupported system: ${stdenv.hostPlatform.system}";
 
   desktopItem = makeDesktopItem {
```

File: pkgs/applications/office/jameica/default.nix [hunk 1]
```diff
@@ -101,7 +101,7 @@ stdenv.mkDerivation rec {
       binaryBytecode # source bundles dependencies as jars
     ];
     license = licenses.gpl2Plus;
-    platforms = [ "x86_64-linux" "i686-linux" "x86_64-darwin" "aarch64-linux" ];
+    platforms = [ "x86_64-linux" "i686-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin" ];
```
Comment (DontEatOreo) [inline pkgs/applications/office/jameica/default.nix:104]:
```suggestion
    platforms = platforms.unix;
```

**Review by 4evy — CHANGES_REQUESTED** [review]

**Comment by DontEatOreo** [comment]
Please rebase and reword your commit to follow [Commit conventions](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions)

In your case you can reword your commit to:

```
jameica: add `arch64-darwin` support
```

**Comment by fbettag** [comment]
@DontEatOreo thanks for taking the time!

**Comment by fbettag** [comment]
i was wondeering if i can also get .app from the dmg so it gets nicely linked into ~/Applications/Nix\ Home\ Manager

**Comment by ryand56** [comment]
Also change base branch to master.

**Comment by fbettag** [comment]
gonna retry on a different pr.. this went sideways fast.

**Pull request closed without merging**